qemu/tests/iperf_test.py

```diff
@@ -0,0 +1,173 @@
+            return os.path.join(iperf_source_path, 'src', iperf_bin_name)
+
+    def iperf_start(session, iperf_path, options, catch_data):
+        """Start iperf session, analyze result if catch_data."""
+        iperf_cmd = iperf_path + options
+        try:
+            info_text = "Start iperf session in %s with cmd: %s"
+            if session:
+                logging.info(info_text % ("guest", iperf_cmd))
+                data_info = session.cmd_output(iperf_cmd, timeout=120)
+            else:
+                logging.info(info_text % ("host", iperf_cmd))
+                data_info = process.system_output(iperf_cmd, timeout=120,
+                                                  verbose=False).decode()
+        except Exception as err_msg:
+            logging.error(str(err_msg))
+            test.error("Failed to start iperf session")
+        else:
+            if catch_data:
+                logging.debug("Full connection log:\n%s" % data_info)
+                parallel_cur = len(re.findall(catch_data, data_info))
+                parallel_exp = int(params.get("parallel_num", 0))
+                if not parallel_cur:
+                    test.fail("iperf client not connected to server")
+                elif parallel_exp and parallel_cur != parallel_exp:
+                    test.fail("Number of parallel threads running(%d) is "
+                              "inconsistent with expectations(%d)"
+                              % (parallel_cur, parallel_exp))
+                logging.info("iperf client successfully connected to server")
+
+    def is_iperf_running(name_pattern, session=None):
+        if session:
+            check_iperf_cmd = params["check_iperf_cmd"] % name_pattern
+            status = serial_session.cmd_status(check_iperf_cmd)
+        else:
+            status = process.system("pgrep -f %s" % name_pattern,
+                                    ignore_status=True, verbose=False)
+        return status == 0
+
+    os_type = params["os_type"]
+    login_timeout = int(params.get("login_timeout", 360))
+    fw_stop_cmd = params["fw_stop_cmd"]
+    tmp_dir = params.get("tmp_dir", "/tmp/")
+    iperf_test_duration = int(params["iperf_test_duration"])
+    iperf_deps_dir = data_dir.get_deps_dir("iperf")
+    host_iperf_file = params["host_iperf_file"]
+    guest_iperf_file = params.get('guest_iperf_file', host_iperf_file)
+    host_src_path = os.path.join(iperf_deps_dir, host_iperf_file)
+    guest_src_path = os.path.join(iperf_deps_dir, guest_iperf_file)
+    host_dst_path = tmp_dir + host_iperf_file
+    guest_dst_path = params.get('guest_dst_path', host_dst_path)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    serial_session = vm.wait_for_serial_login(timeout=login_timeout)
+    guest_session = vm.wait_for_login(timeout=login_timeout)
+    guest_session.cmd(fw_stop_cmd, ignore_all_errors=True)
+
+    vm.copy_files_to(guest_src_path, guest_dst_path)
```

Comment:
`copy_files_to` is used to transfer files to remote.  Here is copy file from host to guest as my understanding.

Also,  without param `guest_iperf_file`,  `host_src_path` and `guest_src_path`  are same,  bur it is not a good use in my opinion.
```
 guest_iperf_file = params.get('guest_iperf_file', host_iperf_file)
 host_src_path = os.path.join(iperf_deps_dir, host_iperf_file)
 guest_src_path = os.path.join(iperf_deps_dir, guest_iperf_file)
```

Reply:
when `os_type = linux`, host and guest will use the same source tool code, so `guest_src_path` = `host_src_path`. Therefore, I think we don't need to redefine the guest path.

Reply:
I see `guest_src_path` and `host_src_path` are same with the current design, i have mentioned it in the comment above.  what i mean is,  at least, it should be `vm.copy_files_to(host_src_path, guest_dst_path) `

Because just from code,  we can easily understand it is copy file from guest path A  --> guest path B.